syntax = "proto3";


message UserRequest {
    int32 id = 1;
    string user_name = 2;
    string email = 3;
    string first_name = 4;
    string last_name = 5;
    string ip_address = 6;
    string gender = 7;
    string country = 8;
    string agent = 9;
}


message UserResponse {
    bool status = 1;
    string message = 2;
}

service Users {
    rpc SendUserInfo (UserRequest) returns (UserResponse);
    rpc SendUserInfoClientStream (stream UserRequest) returns (UserResponse);
}
